### ISSUE TITLE
BUG

### ISSUE THREAD

**Issue by Adhaaz** [lamhoang1256/netfilm#20]
Sir @lamhoang1256, The bug when streaming a movie or series happens again. Only shows loading without any video playing 🥹

**Comment by Adhaaz** [comment]
> ![IMG_20230215_161138](https://user-images.githubusercontent.com/75169859/218986096-2aef3a03-5e1c-458c-b231-508251fad923.jpg)
> 
> Sir @lamhoang1256, The bug when streaming a movie or series happens again. Only shows loading without any video playing 🥹

@lamhoang1256 Please, fix it sir 🙏

**Comment by Adhaaz** [comment]
@lamhoang1256

**Comment by Adhaaz** [comment]
> @lamhoang1256

I've tried to fix it myself, but it doesn't work.. Please sir @lamhoang1256 fix this 🙏🫢

**Comment by lamhoang1256** [comment]
@Adhaaz I'm so busy and I don't have time to fix it, I'll try to fix it when I have free time

**Comment by Adhaaz** [comment]
When will it be fixed sir @lamhoang1256 ? I've tried to fix it myself but it didn't work 😀

**Comment by lamhoang1256** [comment]
@Adhaaz When the work is done, I will fix it soon

**Comment by Adhaaz** [comment]
Okay sir @lamhoang1256 , please fix it as soon as possible sir 🫢 This weekend can the fix be completed sir? ☺️😀

**Comment by lamhoang1256** [comment]
@Adhaaz I don't promise, when I have free time I will fix it soon

**Comment by Adhaaz** [comment]
> @Adhaaz I don't promise, when I have free time I will fix it soon

Okay sir, I'll wait 😀 but don't take too long sir 😂 I've missed a lot of movie episodes sir 🛺 Since this Bug appeared 🫢

**Comment by Adhaaz** [comment]
Come on, sir @lamhoang1256 🙏 Fix this bug asap.. We are all waiting for u to fix this bug

**Comment by Adhaaz** [comment]
> > Come on, sir @lamhoang1256 🙏 Fix this bug asap.. We are all waiting for u to fix this bug
> 
> @Adhaaz give him a time mfs, you misunderstand what hobby app is... i'm pretty sure he's doing his best to fix this. its hard to fix loklok because we can't interact with API directly. rather than just complaining with app you have zero contribute with it, you better help him or FUCK OFF...!!!

👌

**Comment by Danny-Nunez** [comment]
I would love to help but i can't read the new api loklok is using. If anyone has the new api url it would be helpful. I checked cloud stream but they have encoded the api.

**Comment by Adhaaz** [comment]
Yes, sir @lamhoang1256 , I've fixed the API a bit, and managed to overcome the 404 Not Found Page. However, MediaUrl data still cannot be retrieved by the API, only the Qualities data, How do retrieve the MediaUrl data, sir? Maybe I can fix it myself.. Please help, sir 🫢

**Comment by lamhoang1256** [comment]
@Adhaaz The problem is still that due to we not providing the token and secret keys for Loklok, the authKey of the video .m3u8 is invalid, leading to the error cors can't be played. I tried to spend a lot of time fixing it but it's not easy to get it

**Comment by Adhaaz** [comment]
> @Adhaaz The problem is still that due to we not providing the token and secret keys for Loklok, the authKey of the video .m3u8 is invalid, leading to the error cors can't be played. I tried to spend a lot of time fixing it but it's not easy to get it

Ouh like that, But why can Cloudstream get a token, the secret key of Api Loklok, sir @lamhoang1256 ? Cloudstream can actually play files from the Api Loklok 🤔

**Comment by lamhoang1256** [comment]
@Adhaaz If Cloudstream can actually play files video, you can refer to Cloudstream and fix the error

**Comment by Adhaaz** [comment]
> @Adhaaz If Cloudstream can actually play files video, you can refer to Cloudstream and fix the error

Sir, I just tried embed player from gdriveplayer.to and Player embed this can block access from Loklok sir, can you change the player to embed from player gdriveplayer.to

http://gdriveplayer.to/embed2.php?link=https://akm-cdn-play.loklok.tv/001d8fc0ba2f71ed804436a5e8aa0102/2f64b6bdc25d4ae0b01eea35704dc427-c4197002370b2b48598a4be5d5d0fbde-ld.m3u8?hdnts=exp=1677939075-acl=/*-hmac=fb6f8587076bba250a64115ffe08d0853824be7d04d67eaa46b799d91a0ad496

**Comment by lamhoang1256** [comment]
@Adhaaz Very cool with this discovery bro, but I don't want to add embed  gdriveplayer.to to my web because
1. It has a lot of gdriveplayer ads when using it
2. Can't use subtitles
3. It's very difficult to custom embed to player

**Comment by Adhaaz** [comment]
Yeah sir @lamhoang1256,  I think u need to change the Video Player sir, because I think react-tuby and react-hls-player are hard to break access from Loklok sir... Maybe there is some code that doesn't play through in react-hls-player 🤔

**Comment by Adhaaz** [comment]
> @Adhaaz Very cool with this discovery bro, but I don't want to add embed gdriveplayer.to to my web because
> 
> 1. It has a lot of gdriveplayer ads when using it
> 2. Can't use subtitles
> 3. It's very difficult to custom embed to player

Yes sir @lamhoang1256, I think u need to change the Video Player sir, because I think react-tuby and react-hls-player are difficult to penetrate CDN access from Loklok sir ... There may be some code that cannot be played in react-hls-player 🤔

**Comment by Adhaaz** [comment]
> @Adhaaz Very cool with this discovery bro, but I don't want to add embed gdriveplayer.to to my web because
> 
> 1. It has a lot of gdriveplayer ads when using it
> 2. Can't use subtitles
> 3. It's very difficult to custom embed to player
![IMG_20230313_120126](https://user-images.githubusercontent.com/75169859/224614082-ad49be15-2251-48de-a352-85b866440d4b.png)

I have managed to fix all the bugs on the script code sir @lamhoang1256 , but I encountered one obstacle... The problem is that the video cannot be played in some browsers, such as Chrome, Mozilla, or Opera. But the video can still be played in some browsers. But it can not be played on the Browser I mentioned above. Is there a way to solve that problem, Sir @lamhoang1256 ?
